java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/ArrowFlightJdbcTime.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.driver.jdbc;
+
+import static org.apache.calcite.avatica.util.DateTimeUtils.MILLIS_PER_DAY;
+
+import java.sql.Time;
+import java.time.LocalTime;
+import java.time.temporal.ChronoField;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.arrow.util.VisibleForTesting;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Wrapper class for Time objects to include the milliseconds part in ISO 8601 format in this#toString.
+ */
+public class ArrowFlightJdbcTime extends Time {
+  private static final List<String> LEADING_ZEROES = ImmutableList.of("", "0", "00");
+
+  // Desired length of the millisecond portion should be 3
+  private static final int DESIRED_MILLIS_LENGTH = 3;
+
+  // Millis of the date time object.
+  private final int millisOfSecond;
+
+  /**
+   * Constructs this object based on epoch millis.
+   *
+   * @param milliseconds milliseconds representing Time.
+   */
+  public ArrowFlightJdbcTime(long milliseconds) {
+    super(milliseconds);
+    millisOfSecond = getMillisOfSecond(milliseconds);
+  }
+
+  @VisibleForTesting
+  ArrowFlightJdbcTime(LocalTime time) {
+    // Although the constructor is deprecated, this is the exact same code as Time#valueOf(LocalTime)
+    super(time.getHour(), time.getMinute(), time.getSecond());
+    millisOfSecond = time.get(ChronoField.MILLI_OF_SECOND);
+  }
+
+  private int getMillisOfSecond(long milliseconds) {
+    final int millisOfSecond;
```

Comment:
Can you change this variable names? They are really similar to each other which confuse reading the code